astropy/io/fits/hdu/hdulist.py

```diff
@@ -27,7 +27,7 @@
 from ....extern.six.moves import range
 
 
-def fitsopen(name, mode='readonly', memmap=None, save_backup=False,
+def fitsopen(name, mode=None, memmap=None, save_backup=False,
```

Comment:
Ok I think I know, because it skips the fileobj mode detection ? There is something weird here, if FITS mode is `readonly` but you pass a file handle with `w` it is not detected.